### PR TITLE
Clean up sc-network

### PR DESCRIPTION
**Pull request by liuchengxu**
This PR is mainly about cleaning up `sc-network`. No logical changes, all trivial improvements:

- Avoid using `clone()` for the Copy type like `PeerId`.
- Use `find_map` for `filter_map` and `next`.
- Use `Self`.
- Use `matches!` if suitable.
- Make the log format look a bit more consistent at least per file.

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: client/network/src/service.rs [hunk 1]
```diff
@@ -201,12 +201,7 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
 			params.chain.clone(),
 			params.protocol_id.clone(),
 			&params.network_config,
-			iter::once(Vec::new())
-				.chain(
-					(0..params.network_config.extra_sets.len() - 1)
-						.map(|_| default_notif_handshake_message.clone()),
-				)
-				.collect(),
+			vec![default_notif_handshake_message; params.network_config.extra_sets.len()],
```
Comment (kpp) [inline client/network/src/service.rs:204]: Well, no. elems[0] should be an empty Vec.

Comment (kpp) [reply client/network/src/service.rs:204]: Try this:

```
fn main() {
    core::iter::once(Vec::<()>::new()).inspect(|x| {dbg!(&x);}).collect::<Vec<_>>();
}
```

**Review by kpp — CHANGES_REQUESTED** [review]
There is an issue with vec![default_notif_handshake_message].

File: client/network/src/service.rs (hunk 1 above)
Comment (liuchengxu) [reply client/network/src/service.rs]: Reverted. Interesting, didn't realize the code is not equivalent as before,  `[[]]` is not `[]`, https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=12e10b489de9b92a423279541eb5e317 . Do you mind sharing more about why elems[0] has to be an empty Vec?

File: client/network/src/service.rs [hunk 2]
```diff
@@ -201,7 +201,12 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
 			params.chain.clone(),
 			params.protocol_id.clone(),
 			&params.network_config,
-			vec![default_notif_handshake_message; params.network_config.extra_sets.len()],
+			iter::once(Vec::new())
+				.chain(
+					(0..params.network_config.extra_sets.len() - 1)
```
Comment (kpp) [inline client/network/src/service.rs]: `.chain(vec![default_notif_handshake_message; params.network_config.extra_sets.len()-1])` should also work

**Review by kpp — APPROVED** [review]

**Pull request merged**